pkg/events/derive/symbols_loaded_test.go

```diff
@@ -0,0 +1,194 @@
+package derive
+
+import (
+	"testing"
+
+	"github.com/aquasecurity/tracee/pkg/utils/sharedobjs"
+	"github.com/aquasecurity/tracee/types/trace"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type soInstance struct {
+	info sharedobjs.ObjInfo
+	syms []string
+}
+
+type symbolsLoaderMock struct {
+	cache map[sharedobjs.ObjInfo]map[string]bool
+}
+
+func initLoaderMock() symbolsLoaderMock {
+	return symbolsLoaderMock{cache: make(map[sharedobjs.ObjInfo]map[string]bool)}
+}
+
+func (loader symbolsLoaderMock) GetDynamicSymbols(info sharedobjs.ObjInfo) (map[string]bool, error) {
+	return loader.cache[info], nil
+}
+
+func (loader symbolsLoaderMock) GetExportedSymbols(info sharedobjs.ObjInfo) (map[string]bool, error) {
+	return loader.cache[info], nil
+}
+
+func (loader symbolsLoaderMock) GetImportedSymbols(info sharedobjs.ObjInfo) (map[string]bool, error) {
+	return nil, nil
+}
+
+func (loader symbolsLoaderMock) addSOSymbols(info soInstance) {
+	symsMap := make(map[string]bool)
+	for _, s := range info.syms {
+		symsMap[s] = true
+	}
+	loader.cache[info.info] = symsMap
+}
+
+func generateSOLoadedEvent(pid int, so sharedobjs.ObjInfo) trace.Event {
+	return trace.Event{
+		EventName:     "shared_object_loaded",
+		EventID:       1036,
+		HostProcessID: pid,
+		ProcessID:     pid,
+		Args: []trace.Argument{
+			{ArgMeta: trace.ArgMeta{Type: "const char*", Name: "pathname"}, Value: so.Path},
+			{ArgMeta: trace.ArgMeta{Type: "int", Name: "flags"}, Value: 0},
+			{ArgMeta: trace.ArgMeta{Type: "dev_t", Name: "dev"}, Value: so.Id.Device},
+			{ArgMeta: trace.ArgMeta{Type: "unsigned long", Name: "inode"}, Value: so.Id.Inode},
+			{ArgMeta: trace.ArgMeta{Type: "unsigned long", Name: "ctime"}, Value: so.Id.Ctime},
+		},
+	}
+}
+
+func TestDeriveSharedObjectExportWatchedSymbols(t *testing.T) {
+	type testSO struct {
+		so                             soInstance
+		expectedWatchedExportedSymbols []string
```

Comment:
Wouldn't it make more sense if `expectedWatchedExportedSymbols` was in the testCase struct and not the testSo struct? Then you could just have an `so` and `expected...` field in the testCase, would look better imo.